pkg/daemon/cleanup/cleanup.go

```diff
@@ -72,47 +82,99 @@ func Deconfigure(cfg *rest.Config, ctx context.Context, configMapNamespace strin
 	return nil
 }
 
-// getServicesToRemove returns a list of services that should be removed as part of the cleanup process
-// returns the merged Data of the latest ConfigMap, and the ConfigMap specified by the node's version annotation
-func getServicesToRemove(ctx context.Context, cli client.Client, node *core.Node, configMapNamespace string) ([]servicescm.Service, error) {
-	// get data from the latest services ConfigMap
-	latestCM, err := servicescm.GetLatest(cli, ctx, configMapNamespace)
+// ensureEnvVarsAreRemoved removes all WICD configured ENV variables from this instance
+func ensureEnvVarsAreRemoved(ctx context.Context, cli client.Client, node *core.Node,
+	configMapNamespace string, latestCM *core.ConfigMap) error {
+	var cmData *servicescm.Data
+	var err error
+	versionCM, err := getVersionCM(ctx, cli, node, configMapNamespace)
 	if err != nil {
-		return nil, fmt.Errorf("cannot get latest services ConfigMap from namespace %s: %w", configMapNamespace, err)
+		klog.Infof("error getting services ConfigMap associated with version annotation, "+
+			"falling back to use latest services ConfigMap: %s", err)
+		cmData, err = servicescm.Parse(latestCM.Data)
+	} else {
+		cmData, err = servicescm.Parse(versionCM.Data)
 	}
-	latestCMData, err := servicescm.Parse(latestCM.Data)
 	if err != nil {
-		return nil, err
+		return err
+	}
+	if len(cmData.EnvironmentVars) == 0 {
+		// no-op
+		return nil
+	}
+	// systemEnvVarRegistryPath is where system level environment variables are stored in the Windows OS
+	const systemEnvVarRegistryPath = `SYSTEM\CurrentControlSet\Control\Session Manager\Environment`
```

Comment:
same constant exists within a func somewhere in `controller` package. consider making it global over there and referencing that here